lib/active_record/archiver/store/s3.rb

```diff
@@ -10,23 +10,15 @@ def initialize(args)
       @path     = args.fetch('path', '%Y/%m/%d/%s.%6N.json.gz')
       @options  = args.fetch('options', {})
 
-      @client = Aws::S3::Client.new(@options.deep_symbolize_keys)
+      create_client
```

Comment:
This is an opaque assignment which works due to a side-effect of that method. If you want to use `create_client` then at least let's go with
```suggestion
      @client = create_client
```